test/relayserver/ServerConfigParams.test.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  entriesToObj,
+  filterMembers,
+  filterType,
+  parseServerConfig,
+  resolveServerConfig
+} from '../../src/relayserver/ServerConfigParams'
+import * as fs from 'fs'
+import { expectRevert } from '@openzeppelin/test-helpers'
+import {
+  VersionOracleInstance
+} from '../../types/truffle-contracts'
+import { string32 } from '../../src/common/VersionOracle'
+
+require('source-map-support').install({ errorFormatterForce: true })
+const VersionOracleContract = artifacts.require('VersionOracle')
+
+function expectThrow (func: () => void, match: string): void {
+  try {
+    func()
+  } catch (e) {
+    assert.include(e.toString(), match)
+    return
+  }
+  assert.fail('expected to fail with: ' + match)
+}
+
+function addr (n: number): string {
+  return '0x'.padEnd(42, n.toString())
+}
+
+context('#ServerConfigParams', () => {
+  context('utils', () => {
+    it('#filterType', () => {
+      assert.deepEqual(
+        filterType({ a: 'number', b: 'string', c: 'number' }, 'number'),
+        ['a', 'c'])
+    })
+    it('#entriesToObj', () => {
+      const a = { x: 1, y: 2, z: { a: 11, b: 22 } }
+      assert.deepEqual(a, entriesToObj(Object.entries(a)))
+    })
+
+    it('#filterMembers', () => {
+      const a = { x: 1, y: 2, z: 3 }
+      const config = { x: 'number', y: 'string' }
+
+      assert.deepEqual(filterMembers(a, config), { x: 1, y: 2 })
+    })
+  })
+
+  context('#parseServerConfig', () => {
+    const tmpConfigfile = '/tmp/test.configfile.tmp'
+    after(() => {
+      if (fs.existsSync(tmpConfigfile)) {
+        fs.unlinkSync(tmpConfigfile)
+      }
+    })
+    it('should parse command line params', function () {
+      assert.deepEqual(
+        parseServerConfig(['--devMode=true', '--relayHubAddress=123'], {}),
+        { devMode: true, relayHubAddress: '123' })
+    })
+    it('should use env as defaults', function () {
```

Comment:
While this test somehow check it, precedence of `cmdline > env var >  config file` is not explicitly tested.

Reply:
added explicit override test.